nixos/modules/system/boot/systemd.nix

```diff
@@ -411,30 +411,6 @@ in
   ###### implementation
 
   config = {
-
-    warnings = concatLists (
-      mapAttrsToList
-        (name: service:
-          let
-            type = service.serviceConfig.Type or "";
-            restart = service.serviceConfig.Restart or "no";
-            hasDeprecated = builtins.hasAttr "StartLimitInterval" service.serviceConfig;
-          in
-            concatLists [
-              (optional (type == "oneshot" && (restart == "always" || restart == "on-success"))
-                "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'"
-              )
-              (optional hasDeprecated
-                "Service '${name}.service' uses the attribute 'StartLimitInterval' in the Service section, which is deprecated. See https://github.com/NixOS/nixpkgs/issues/45786."
```

Comment:
I think this warning can be removed without substitution, because man systemd.service do not document this option and we do not check for non existent options.